lib/std/compress/deflate/compressor.zig

```diff
@@ -0,0 +1,1133 @@
+const std = @import("std");
+const fmt = std.fmt;
+const io = std.io;
+const math = std.math;
+const mem = std.mem;
+
+const Allocator = std.mem.Allocator;
+
+const deflate_const = @import("deflate_const.zig");
+const fast = @import("deflate_fast.zig");
+const hm_bw = @import("huffman_bit_writer.zig");
+const mu = @import("mem_utils.zig");
+const token = @import("token.zig");
+
+pub const Compression = enum(i5) {
+    // huffman_only disables Lempel-Ziv match searching and only performs Huffman
+    // entropy encoding. This mode is useful in compressing data that has
+    // already been compressed with an LZ style algorithm (e.g. Snappy or LZ4)
+    // that lacks an entropy encoder. Compression gains are achieved when
+    // certain bytes in the input stream occur more frequently than others.
+    //
+    // Note that huffman_only produces a compressed output that is
+    // RFC 1951 compliant. That is, any valid DEFLATE decompressor will
+    // continue to be able to decompress this output.
+    huffman_only = -2,
+    default_compression = -1,
+    no_compression = 0,
+    best_speed = 1,
+    level_2 = 2,
+    level_3 = 3,
+    level_4 = 4,
+    level_5 = 5,
+    level_6 = 6,
+    level_7 = 7,
+    level_8 = 8,
+    best_compression = 9,
+};
+pub const Compressions = [_]Compression{
+    Compression.huffman_only,
+    Compression.default_compression,
```

Comment:
Could use enum literals here and many other places in this code.
```suggestion
    .huffman_only,
    .default_compression,
```

Reply:
used enum literals